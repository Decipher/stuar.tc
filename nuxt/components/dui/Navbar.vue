<template>
  <div class="bg-white navbar sticky text-white-content z-50">
    <div class="flex-1 px-2 mx-2">
      <NuxtLink to="/" class="text-lg font-bold" v-text="title" />
    </div>
    <div class="flex-none hidden px-2 mx-2 md:flex">
      <div class="flex items-stretch">
        <!-- Menu Items -->
      </div>
    </div>
    <div class="flex-none">
      <!-- @TODO - Make this work -->
      <!-- <button class="btn btn-square btn-ghost">
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          class="inline-block w-6 h-6 stroke-current"
        >
          <path
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            d="M4 6h16M4 12h16M4 18h16"
          ></path>
        </svg>
      </button> -->
    </div>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      required: true,
    },
  },
}
</script>
