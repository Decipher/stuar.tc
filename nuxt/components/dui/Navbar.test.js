import { mount } from '@vue/test-utils'
import Navbar from './Navbar.vue'

const stubs = ['NuxtLink']

describe('Navbar', () => {
  test('is a Vue instance', () => {
    const props = { title: 'Stuart Clark' }
    const wrapper = mount(Navbar, { props, stubs })
    expect(wrapper.vm).toBeTruthy()

    // Expect HTML to match snapshot.
    expect(wrapper.html()).toMatchSnapshot()
  })
})
