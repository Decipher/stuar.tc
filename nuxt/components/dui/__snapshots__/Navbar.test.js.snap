// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar is a Vue instance 1`] = `
"<div class=\\"bg-white navbar sticky text-white-content z-50\\">
  <div class=\\"flex-1 px-2 mx-2\\">
    <nuxtlink-stub to=\\"/\\" class=\\"text-lg font-bold\\"></nuxtlink-stub>
  </div>
  <div class=\\"flex-none hidden px-2 mx-2 md:flex\\">
    <div class=\\"flex items-stretch\\"></div>
  </div>
  <div class=\\"flex-none\\"></div>
</div>"
`;
